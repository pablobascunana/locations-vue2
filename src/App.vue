<template>
  <v-app id="app">
    <v-main>
      <AppBar v-if="!isLogin"></AppBar>
      <v-container fill-height> 
        <v-row>
          <v-col cols="12">
            <router-view />
          </v-col>
        </v-row>
        <SnackBar
          :color="snackBarColor"
          :message="snackBarMessage"
          v-if="showSnackBar"
          @close="snackBarHandler">
        </SnackBar>
        <LogoutDialog></LogoutDialog>
      </v-container>
    </v-main>
  </v-app>
</template>

<script>
import { mapState } from 'vuex';
export default {
  name: 'app',
  components: {
    AppBar: () => import('@/components/AppBar'),
    SnackBar: () => import('@/components/wrappers/SnackBar'),
    LogoutDialog: () => import('@/components/dialogs/Logout')
  },
  computed: {
    ...mapState('application', ['isLogin'])
  },
  mounted() {
    this.EventBus.$on('showSnackBar', (data) => {
      this.fillAndShowSnackBar(data);
    });
  },
  data() {
    return {
      showSnackBar: false,
      snackBarColor: '',
      snackBarMessage: ''
    }
  },
  methods: {
    fillAndShowSnackBar(data) {
      this.snackBarColor = typeof data.snackBarColor !== 'undefined' ? data.snackBarColor : 'success';
      this.snackBarMessage = data.snackBarMessage;
      this.showSnackBar = true;
    },
    snackBarHandler() {
      this.showSnackBar = false;
    },
  }
}
</script>
