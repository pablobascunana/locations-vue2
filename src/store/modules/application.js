function initialApplicationState() {
  return {
    isLogin: true,
    accessToken: undefined,
    refreshToken: undefined,
    addLocationDialogVisible: false,
    removeLocationDialogVisible: false
  }
}

const application = {
  namespaced: true,
  state: initialApplicationState,
  mutations: {
    resetApplicationStore(state) {
      const initState = initialApplicationState();
      Object.keys(initState).forEach(key => {
        state[key] = initState[key];
      })
    },
    isLogin(state, visible) {
      state.isLogin = visible;
    },
    setAccessToken(state, token) {
      state.accessToken = `Bearer ${token}`;
    },
    setRefreshToken(state, token) {
      state.refreshToken = token;
    },
    isAddLocationDialogVisible(state, visible) {
      state.addLocationDialogVisible = visible;
    },
    isRemoveLocationDialogVisible(state, visible) {
      state.removeLocationDialogVisible = visible
    }
  }
};

export default application;
  