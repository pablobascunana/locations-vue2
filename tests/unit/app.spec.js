import { createLocalVue, shallowMount } from '@vue/test-utils'

import Vuetify from 'vuetify'
import Vuex from 'vuex'

import application from '@/store/modules/application.js'

import App from '@/App.vue'

import * as appMock from './mocks/app.mock.js'

const Plugins = {
    install(Vue) {
      Vue.prototype.EventBus = new Vue();
    }
};

const localVue = createLocalVue();
localVue.use(Plugins);
localVue.use(Vuex);
let vuetify = new Vuetify();


describe('App', () => {
    let wrapper;
    let store;
    let state = {};
    let actions = {};
    let modules = {
        application
    };
    const mocks = {
        EventBus: {
            $on: jest.fn()
        },
        $t: () => {'some specific text in any language'}
    };
    beforeEach(() => {
        store = new Vuex.Store({ state, actions, modules });

        wrapper = shallowMount(App, {
            localVue,
            vuetify,
            store,
            mocks
        });
    });

    it("Should match snapshot", () => {
        expect(wrapper).toMatchSnapshot()    
    });

    it('Renders a vue instance', () => {
        expect(wrapper.isVueInstance()).toBe(true);
    });

    it('Renders AppBar component', () => {
        wrapper.vm.$store.commit('application/isLogin', false);
    });

    it('Call to fill and show snackBar', () => {
        const fillAndShowSnackBar = jest.spyOn(wrapper.vm, 'fillAndShowSnackBar');
        wrapper.vm.fillAndShowSnackBar(appMock.SUCCESS_SNACK_BAR);
        wrapper.vm.fillAndShowSnackBar(appMock.ERROR_SNACK_BAR);
        expect(fillAndShowSnackBar).toBeCalled();
    });

    it('Call to snackBar handler', () => {
        const snackBarHandlerFn = jest.spyOn(wrapper.vm, 'snackBarHandler');
        wrapper.vm.snackBarHandler();
        expect(snackBarHandlerFn).toBeCalled();
        expect(wrapper.vm.showSnackBar).toBe(false);
    });
});