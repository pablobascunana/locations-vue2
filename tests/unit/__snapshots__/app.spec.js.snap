// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match snapshot 1`] = `
<v-app id="app">
  <v-main>
    <!---->
    <v-container fill-height="">
      <v-row>
        <v-col cols="12">
          <router-view></router-view>
        </v-col>
      </v-row>
      <!---->
      <v-dialog max-width="500px">
        <v-card>
          <v-card-title>

            <v-spacer></v-spacer>
            <v-icon aria-label="Close">mdi-close</v-icon>
          </v-card-title>
          <v-card-text class="text-center mt-10">
            <p></p>
          </v-card-text>
          <v-card-actions>
            <v-spacer></v-spacer>
            <v-btn id="logout" color="primary" text="">
            </v-btn>
          </v-card-actions>
        </v-card>
      </v-dialog>
    </v-container>
  </v-main>
</v-app>
`;
